R/utils.R

```diff
@@ -372,16 +372,74 @@ markdown_to_text <- function(text) {
     unname()
 }
 
-# Get prepending and appending text based on a simple pattern
-get_pre_post_txt <- function(pattern) {
+#' Handle formatting of a pattern in a \code{fmt_*()} function
+#'
+#' Within the context of a \code{fmt_*()} function, we always have the
+#' single-length character vector of \code{pattern} available to describe a
+#' final decoration of the formatted values. We use \pkg{glue}'s semantics here
+#' (and literally, it's \code{glue::glue()}) and reserve \code{x} to be the
+#' formatted values. We can use \code{x} multiple times in the pattern or even
+#' variables in global scope or R statements within curly braces.
+#' @param pattern An expression string that is the primary input to
+#'   \code{glue::glue()}). The variable \code{x} is reserved and represents the
+#'   formatted values.
+#' @param values The values (as a character vector) that are formatting within
+#'   the \code{fmt_*()} function.
+#' @importFrom glue glue
+#' @noRd
+apply_pattern_fmt_x <- function(pattern,
+                                values) {
 
-  prefix <- strsplit(pattern, "\\{x\\}")[[1]][1]
-  suffix <- strsplit(pattern, "\\{x\\}")[[1]][2]
+  glue::glue(pattern, x = values) %>% as.character()
```

Comment:
Hmmm, WTF, glue? The presence of a newline can affect whitespace trimming behavior??

```
> glue::glue("   {'1'}   ") %>% as.character()
[1] "   1   "
> glue::glue("   {'1'}\n   ") %>% as.character()
[1] "1"
```

Reply:
I think this is a case (aside from the other point) for not using glue.